File: gitlfs-common/src/test/java/ru/bozaro/gitlfs/common/data/CreateLockReqTest.java
package ru.bozaro.gitlfs.common.data;

import org.testng.Assert;
import org.testng.annotations.Test;

import java.io.IOException;

public final class CreateLockReqTest {
  @Test
  public void parse01() throws IOException {
    final CreateLockReq data = SerializeTester.deserialize("create-lock-req-01.json", CreateLockReq.class);
    Assert.assertNotNull(data);
    Assert.assertEquals(data.getPath(), "foo/bar.zip");

    Assert.assertNotNull(data.getRef());
    Assert.assertEquals(data.getRef().getName(), "refs/heads/my-feature");
  }
}

File: gitlfs-common/src/test/kotlin/ru/bozaro/gitlfs/common/data/CreateLockReqTest.kt
package ru.bozaro.gitlfs.common.data

import org.testng.Assert
import org.testng.annotations.Test
import java.io.IOException

class CreateLockReqTest {
    @Test
    @Throws(IOException::class)
    fun parse01() {
        val data: CreateLockReq = SerializeTester.deserialize("create-lock-req-01.json", CreateLockReq::class.java)
        Assert.assertNotNull(data)
        Assert.assertEquals(data.path, "foo/bar.zip")
        Assert.assertNotNull(data.ref)
        Assert.assertEquals(data.ref!!.name, "refs/heads/my-feature")
    }
}
